### PR TITLE
4 authentification, 5 Database, Display Events , Event Card

### PR DESCRIPTION
**Pull request by Khalil-NOUI**
-initiate Database.
-enable Authentication.
-display event.
-create card event.
-Sign Up and Sign In forms.

Screenshot 2023-10-18s

Screenshot 2023-10-18

2023-10-18 Screenshot

Screenshot 2023-10-18l

### CODE REVIEW

File: src/components/footer/Footer.jsx [hunk 1]
```diff
@@ -23,7 +23,7 @@ function Footer() {
                     <div className='items-center justify-center text-center content-center p-2'>
                         <Link
                             className='items-center justify-center content-center p-2'
-                            href='/events'
+                            href='../../pages/events/'
```
Comment (HachemBouhadede) [inline src/components/footer/Footer.jsx:26]: u don't includes the word poages khalil ! just a href directly to path

File: src/lib/firebase/controller.ts
Comment (HachemBouhadede) [inline src/lib/firebase/controller.ts:1]: this is what you were assigned initially

File: src/lib/firebase/firebase.ts
Comment (HachemBouhadede) [inline src/lib/firebase/firebase.ts:1]: along with this, i don't know why you you used type script  ! but doesn't matter as long as it works
great work <3

File: src/pages/authentication/signIn.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                );
+            })
+            .catch((err) => {
+                const errCode = err.code;
+                const errMsg = err.message;
+                console.log(`${errMsg} and code is : ${errCode}`);
+            });
+    }
+
+    function handleSignOut(e) {
```
Comment (HachemBouhadede) [inline src/pages/authentication/signIn.js:30]: this one should be add in navbar as on click for sign out button

File: src/pages/authentication/signIn.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
+import { signInWithEmailAndPassword, signOut } from "firebase/auth";
+import Link from "next/link";
+import React from "react";
+
+import { auth } from "@/lib/firebase/controller";
+
+export default function SignInOut() {
```
Comment (HachemBouhadede) [inline src/pages/authentication/signIn.js:7]: i suggest you rename function name

File: src/pages/authentication/signIn.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+                                className='block text-sm font-semibold text-gray-800'
+                            >
+                                Password
+                            </label>
+                            <input
+                                type='password'
+                                className='block w-full px-4 py-2 mt-2 text-gray-700 bg-white border rounded-md focus:border-gray-400 focus:ring-gray-300 focus:outline-none focus:ring focus:ring-opacity-40'
+                            />
+                        </div>
+                        <p className='text-xs text-gray-800 font-bold'>
```
Comment (HachemBouhadede) [inline src/pages/authentication/signIn.js:80]: shouldn't be there a functionalmity for this ?? or is just temporary inavailble

File: src/components/eventsPageComponents/EventList.jsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+                Our Events
+            </h1>
+            {events.map((event) => {
+                return <EventCard key={event.id} TheEvent={event} />;
+            })}
+        </div>
+    );
+}
+
+export default EventList;
```
Comment (HachemBouhadede) [inline src/components/eventsPageComponents/EventList.jsx:32]: well done on this one !

File: src/components/navbar/Navbar.jsx [hunk 6]
```diff
@@ -81,12 +81,12 @@ const Navbar = () => {
...
                                             <>
-                                                <Link href='/signin'>
+                                                <Link href='../../pages/signIn/signIn'>
                                                     <button className='w-full p-2 flex items-center justify-center hover:bg-amber-300 hover:text-white rounded-lg'>
                                                         {t("Sign In")}
                                                     </button>
                                                 </Link>
-                                                <Link href='/signup'>
+                                                <Link href='../../pages/signUp/signUp'>
                                                     <button className='w-full p-2 flex items-center justify-center hover:bg-amber-300 hover:text-white rounded-lg'>
```
Comment (HachemBouhadede) [inline src/components/navbar/Navbar.jsx:90]: same for this as footer

File: src/pages/authentication/signIn.js [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+                            </button>
+                        </div>
+                    </form>
+
+                    <p className='mt-2 text-xs text-center text-gray-700'>
+                        {" "}
+                        Become a member?{" "}
+                        <Link
+                            className='font-medium text-gray-600 hover:underline'
+                            href='/signUp/signUp'
```
Comment (HachemBouhadede) [inline src/pages/authentication/signIn.js:101]: i believe this is temporary !

File: src/pages/authentication/signUp.js [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+                        </div>
+                    </form>
+
+                    <p className='mt-2 text-xs text-center text-gray-700'>
+                        {" "}
+                        Already a member?{" "}
+                        <Link
+                            className='font-medium text-gray-600 hover:underline'
+                            href='/signIn'
+                        >
```
Comment (HachemBouhadede) [inline src/pages/authentication/signUp.js:115]: again i'm assuming href is temporary

**Review by HachemBouhadede — CHANGES_REQUESTED** [review]
@Khalil-NOUI get these asap

- [ ] re-update to get the latest develop version 
- [ ] add right path to hrefs
- [ ] add signout function to nav, we can do that togathger if you wish

note: you'e done way more then you were initially assigned, make small pull requests not huge once

**Review by Khalil-NOUI — COMMENTED** [review]
thank you for the the detailed reviews, most acccurate will be dealt with ASAP

**Comment by Khalil-NOUI** [comment]
a newer PR will present the updated changes

**Pull request closed without merging**